src/Datadog.Trace.ClrProfiler.Managed/Integrations/Types/TypeExtensions.cs

```diff
@@ -0,0 +1,29 @@
+namespace Datadog.Trace.ClrProfiler
+{
+    internal static class TypeExtensions
+    {
+        public static System.Type GetInstrumentedType(
+            this object runtimeObject,
+            string instrumentedTypeName)
+        {
+            if (runtimeObject == null)
+            {
+                return null;
+            }
+
+            var currentType = runtimeObject.GetType();
+
+            while (currentType != null)
+            {
+                if ($"{currentType.Namespace}.{currentType.Name}" == instrumentedTypeName)
```

Comment:
I don't think we need this method, but if we keep it, we can use `Type.FullName` instead of concatenating:
```suggestion
                if (currentType.FullName == instrumentedTypeName)
```